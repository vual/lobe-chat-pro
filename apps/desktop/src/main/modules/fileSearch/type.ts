import { FileResult, SearchOptions } from '@/types/fileSearch';

/**
 * File Search Service Implementation Abstract Class
 * Defines the interface that different platform file search implementations need to implement
 */
export abstract class FileSearchImpl {
  /**
   * Perform file search
   * @param options Search options
   * @returns Promise of search result list
   */
  abstract search(options: SearchOptions): Promise<FileResult[]>;

  /**
   * Check search service status
   * @returns Promise indicating if service is available
   */
  abstract checkSearchServiceStatus(): Promise<boolean>;

  /**
   * Update search index
   * @param path Optional specified path
   * @returns Promise indicating operation success
   */
  abstract updateSearchIndex(path?: string): Promise<boolean>;
}
