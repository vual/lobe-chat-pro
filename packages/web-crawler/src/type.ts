export interface CrawlSuccessResult {
  content?: string;
  contentType: 'text' | 'json';
  description?: string;
  length?: number;
  siteName?: string;
  title?: string;
  url: string;
}

export interface CrawlErrorResult {
  content: string;
  errorMessage: string;
  errorType: string;
  url: string;
}

export interface FilterOptions {
  // 是否启用Readability
  enableReadability?: boolean;

  pureText?: boolean;
}

type CrawlImplType = 'naive' | 'jina' | 'browserless' | 'search1api';

type CrawlImplParams<T> = T & {
  filterOptions: FilterOptions;
};

export type CrawlImpl<Params = object> = (
  url: string,
  params: CrawlImplParams<Params>,
) => Promise<CrawlSuccessResult | undefined>;

export interface CrawlUrlRule {
  // 内容过滤配置（可选）
  filterOptions?: FilterOptions;
  impls?: CrawlImplType[];
  // URL匹配模式，仅支持正则表达式
  urlPattern: string;
  // URL转换模板（可选），如果提供则进行URL转换
  urlTransform?: string;
}
