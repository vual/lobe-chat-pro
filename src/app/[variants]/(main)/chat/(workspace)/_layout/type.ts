import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  conversation: ReactNode;
  portal: ReactNode;
  topic: ReactNode;
}
