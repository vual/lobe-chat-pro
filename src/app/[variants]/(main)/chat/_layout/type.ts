import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  session: ReactNode;
}
