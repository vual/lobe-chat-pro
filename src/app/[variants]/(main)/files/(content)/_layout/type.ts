import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  menu: ReactNode;
  modal: ReactNode;
}
