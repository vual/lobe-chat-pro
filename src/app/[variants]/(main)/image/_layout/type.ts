import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  menu: ReactNode;
  topic: ReactNode;
}
