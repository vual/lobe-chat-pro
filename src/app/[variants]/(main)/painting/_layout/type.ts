import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  palette: ReactNode;
}
