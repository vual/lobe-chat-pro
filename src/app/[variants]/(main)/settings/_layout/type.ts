import { ReactNode } from 'react';

export interface LayoutProps {
  category: ReactNode;
  children: ReactNode;
}
