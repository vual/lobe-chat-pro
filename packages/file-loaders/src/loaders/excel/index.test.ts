import path from 'node:path';
import { beforeEach, describe, expect, it } from 'vitest';

import type { FileLoaderInterface } from '../../types';
import { ExcelLoader } from './index';

// 确保你已经在 fixtures 目录下放置了 test.xlsx 文件
// 这个 Excel 文件最好包含多个工作表 (sheets) 以便测试
const fixturePath = (filename: string) => path.join(__dirname, `./fixtures/${filename}`);

let loader: FileLoaderInterface;

const testFile = fixturePath('test.xlsx');
const nonExistentFile = fixturePath('nonexistent.xlsx');

beforeEach(() => {
  loader = new ExcelLoader();
});

describe('ExcelLoader', () => {
  it('should load pages correctly from an Excel file (one page per sheet)', async () => {
    const pages = await loader.loadPages(testFile);
    // Excel 文件有多少个 sheet，就应该有多少个 page
    expect(pages.length).toBeGreaterThan(0);

    // 直接对整个 pages 数组进行快照测试
    expect(pages).toMatchSnapshot();

    // 如果你的 test.xlsx 有多个 sheet，可以添加更多断言
    // 例如检查特定 sheet 的 metadata 中的 sheetName
    // expect(pages[1].metadata.sheetName).toBe('Sheet2');
  });

  it('should aggregate content correctly (joining sheets)', async () => {
    const pages = await loader.loadPages(testFile);
    const content = await loader.aggregateContent(pages);
    // 默认聚合是以换行符连接各 sheet 内容
    expect(content).toMatchSnapshot('aggregated_content');
  });

  it('should handle file read errors in loadPages', async () => {
    const pages = await loader.loadPages(nonExistentFile);
    expect(pages).toHaveLength(1); // 即使失败也返回一个包含错误信息的页面
    expect(pages[0].pageContent).toBe('');
    expect(pages[0].metadata.error).toContain('Failed to load Excel file');
  });
});
