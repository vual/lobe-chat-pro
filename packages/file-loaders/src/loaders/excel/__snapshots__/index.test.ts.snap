// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ExcelLoader > should aggregate content correctly (joining sheets) > aggregated_content 1`] = `
"<sheet name="表1" index="0">
| __EMPTY | 类别 A | 类别 B | __EMPTY_1 | __EMPTY_2 |
| --- | --- | --- | --- | --- |
| 项目 1 | 5 | 7 |  |  |
| 项目 2 | 10 | 8 |  |  |
| 项目 3 | 9 | 15 |  |  |
| 项目 4 | 7 | 12 |  |  |
| 项目 5 | 16 | 21 |  |  |
</sheet>

<sheet name="表2 - 表格 2" index="1">
| __EMPTY | 类别 A | 类别 B | __EMPTY_1 | __EMPTY_2 |
| --- | --- | --- | --- | --- |
| 项目 1 | 5 | 7 |  |  |
| 项目 2 | 10 | 8 |  |  |
| 项目 3 | 9 | 15 |  |  |
| 项目 4 | 7 | 12 |  |  |
| 项目 5 | 16 | 21 |  |  |
</sheet>"
`;

exports[`ExcelLoader > should load pages correctly from an Excel file (one page per sheet) 1`] = `
[
  {
    "charCount": 201,
    "lineCount": 7,
    "metadata": {
      "sheetName": "表1",
    },
    "pageContent": "| __EMPTY | 类别 A | 类别 B | __EMPTY_1 | __EMPTY_2 |
| --- | --- | --- | --- | --- |
| 项目 1 | 5 | 7 |  |  |
| 项目 2 | 10 | 8 |  |  |
| 项目 3 | 9 | 15 |  |  |
| 项目 4 | 7 | 12 |  |  |
| 项目 5 | 16 | 21 |  |  |",
  },
  {
    "charCount": 201,
    "lineCount": 7,
    "metadata": {
      "sheetName": "表2 - 表格 2",
    },
    "pageContent": "| __EMPTY | 类别 A | 类别 B | __EMPTY_1 | __EMPTY_2 |
| --- | --- | --- | --- | --- |
| 项目 1 | 5 | 7 |  |  |
| 项目 2 | 10 | 8 |  |  |
| 项目 3 | 9 | 15 |  |  |
| 项目 4 | 7 | 12 |  |  |
| 项目 5 | 16 | 21 |  |  |",
  },
]
`;
